korangar/src/main.rs

```diff
@@ -938,21 +942,6 @@ impl Client {
                         .cloned()
                         .unwrap();
 
-                    self.map = self
```

Comment:
The self.map information is used in set_position from NetworkEvent::SetPlayerPosition.
If you are outside of the boundary of the start screen map, the games crashes.

Reply:
I see, we sometimes convert one network packet into multiple network events, which is ... suboptimal. Yes, the server ist sending also a position when the login "MapServerLoginSuccessPacket" was received, but every ChangeMapPacketPacket will generate an Event, that contains the map name and the position. I think it's best to alwas only send out one NetworkEvent for one packet.

The old way we did it:

CharacterSelectionSuccessPacket -> NetworkEvent::CharacterSelected -> load player, change map, set player at origin
Connect to Map server -> MapServerLoginSuccessPacket -> NetworkEvent::UpdateClientTick + NetworkEvent::SetPlayerPosition -> Set player at correct position  + update client tick
ChangeMapPacket -> NetworkEvent::ChangeMap -> changes map again, set player at correct position


I propose that we simplify the first sparn proces:


CharacterSelectionSuccessPacket -> NetworkEvent::CharacterSelected -> load player, set player at origin
MapServerLoginSuccessPacket -> NetworkEvent::UpdateClientTick -> update client tick
ChangeMapPacket -> NetworkEvent::ChangeMap -> changes map forst the first time, set player at correct position


We essentially now only allow that one packet can create exactly one network event. We also don't use the map information in the CharacterSelectionSuccessPacket , ignore the position in the MapServerLoginSuccessPacket.
But instead only use the map that the map server tolds us to change into and also the position it gaves us.